### PR TITLE
Validate rewrite annotation

### PR DESCRIPTION
**Pull request by haywoodsh**
### Proposed changes
Validate rewrite annotation to ensure (1) "serviceName" and "rewrite" present, (2) serviceName value in "spec", and (3) rewrite paths start with "/" and not include any whitespace character, "{", "}" or "$".

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: internal/configs/parsing_helpers.go [hunk 1]
```diff
@@ -333,28 +333,38 @@ func parseRewrites(service string) (serviceName string, rewrite string, err erro
 	parts := strings.SplitN(strings.TrimSpace(service), " ", 2)
 
 	if len(parts) != 2 {
-		return "", "", fmt.Errorf("Invalid rewrite format: %s", service)
+		return "", "", fmt.Errorf("'%s' is not valid rewrite format, e.g. 'serviceName=tea-svc rewrite=/'", service)
```
Comment (shaun-nx) [inline internal/configs/parsing_helpers.go:336]: I would change to wording of this to be 'is not a valid'

File: internal/configs/parsing_helpers.go [hunk 2]
```diff
@@ -333,28 +333,38 @@ func parseRewrites(service string) (serviceName string, rewrite string, err erro
...
 
 	rwPathParts := strings.Split(parts[1], "=")
-	if len(rwPathParts) != 2 {
-		return "", "", fmt.Errorf("Invalid rewrite format: %s", rwPathParts)
+	if len(rwPathParts) != 2 || rwPathParts[0] != "rewrite" {
+		return "", "", fmt.Errorf("'%s' is not a valid rewrite path format, e.g. 'rewrite=/tea'", parts[1])
+	}
+
+	if !VerifyPath(rwPathParts[1]) {
+		return "", "", fmt.Errorf("path must start with '/' and must not include any whitespace character, '{', '}' or '$': '%s'", rwPathParts[1])
```
Comment (shaun-nx) [inline internal/configs/parsing_helpers.go:350]: 'P' should be upper case for the word 'path' here

File: internal/k8s/validation_test.go [hunk 3]
```diff
@@ -1951,7 +2141,7 @@ func TestValidateNginxIngressAnnotations(t *testing.T) {
 			appProtectDosEnabled:  false,
 			internalRoutesEnabled: true,
 			expectedErrors: []string{
-				`annotations.nginx.org/rewrites: Invalid value: "not_a_rewrite": must be a semicolon-separated list of rewrites`,
+				`annotations.nginx.org/rewrites: Invalid value: "not_a_rewrite": 'not_a_rewrite' is not valid rewrite format, e.g. 'serviceName=tea-svc rewrite=/'`,
```
Comment (shaun-nx) [inline internal/k8s/validation_test.go:2144]: You will need to update this expected error message if you make the update from my first comment.

**Review by jjngx — APPROVED** [review]
Good job!
As refactoring touches files that use ```fmt.Errorf()``` it may be worth to consider going a step further and implement sentinel errors. This would allow to leverage [error handling and wrapping](https://go.dev/doc/go1.13#error_wrapping) introduced in Go 1.13:
```
errors.Is(err, errType)
errors.As(err, &errType)
```
instead of comparing strings in unit tests.
Just food for thought...

File: internal/k8s/validation_test.go (hunk 3 above)
Comment (haywoodsh) [reply internal/k8s/validation_test.go]: fixed

File: internal/configs/parsing_helpers.go (hunk 2 above)
Comment (haywoodsh) [reply internal/configs/parsing_helpers.go:350]: leaving as it is after discussion

**Review by shaun-nx — APPROVED** [review]

**Review by ciarams87 — APPROVED** [review]

**Pull request merged**